internal/provider/utils/utils.go

```diff
@@ -41,17 +41,24 @@ func HashString(str string) string {
 	return strings.ToLower(fmt.Sprintf("%x", h.Sum(nil)))
 }
 
-// ExpectedContainerPortHashedName returns expected container port name with max length of 15 characters.
+// ExpectedContainerPortName returns expected container port name with max length of 15 characters.
 // If mergedGateways is enabled or listener port name is larger than 15 characters it will return partially hashed name.
 // Listeners on merged gateways have a infraIR port name {GatewayNamespace}/{GatewayName}/{ListenerName}.
-func ExpectedContainerPortHashedName(name string) string {
+func ExpectedContainerPortName(name string) string {
```

Comment:
this logic is looking fairly complicated - the output is different based on length as well as special char `/` tokens
can this look more like
```
func GetHashedContainerPortName(nsName string) string {
	hashedName := HashString(nsName)
	// replace `/` with `-` to create a valid K8s resource name
	resourceName := strings.ReplaceAll(nsName, "/", "-")
	if len(resourceName) > 7 {
		return fmt.Sprintf("%s-%s", resourceName[0:7], hashedName[0:8])
	}
	return fmt.Sprintf("%s-%s", resourceName, hashedName[0:8])
}
```